### PR TITLE
[candi] added ssh allow list to cloud providers

### PR DESCRIPTION
**Pull request by RomanenkoDenys**
## Description

Added ssh allow list to cloud providers terraform security groups.
Providers list:
Azure (standard layout).
GCP (standard, without-nat layouts).
AWS (standard, without-nat, with-nat layouts).
Openstack (simple, simple-with-internal-network, standard, standard-with-no-router layouts).
## Why do we need it, and what problem does it solve?

We want to limit ssh access to cluster nodes only to the allowed list of CIDRs.
## Changelog entries

```changes
section: candi
type: feature
summary: Added ssh allow list to cloud providers terraform security groups.
impact_level: low
```

### CODE REVIEW

**Comment by RomanenkoDenys** [comment]
- [ ] Azure test
- [ ] GCP test
- [ ] AWS test
- [ ] Openstack test

File: ee/candi/cloud-providers/openstack/layouts/standard/base-infrastructure/main.tf [hunk 1]
```diff
@@ -5,6 +5,7 @@ module "network_security" {
   source           = "../../../terraform-modules/network-security"
   prefix           = local.prefix
   remote_ip_prefix = "0.0.0.0/0"
```
Comment (konstantin-axenov) [inline ee/candi/cloud-providers/openstack/layouts/standard/base-infrastructure/main.tf:7]: Can we get rid of `remote_ip_prefix` variable? I think it is confusing if we have `ssh_allow_list`.

Comment (RomanenkoDenys) [reply ee/candi/cloud-providers/openstack/layouts/standard/base-infrastructure/main.tf]: Yes, that's a good point.

**Review by konstantin-axenov — APPROVED** [review]

**Comment by konstantin-axenov** [comment]
@nabokihms Have a look, please.

**Pull request merged**